### ISSUE TITLE
trust data missing for remote repository <my_registry>/<my_image> or remote repository not found: timestamp key trust data unavailable.  Has a notary repository been initialized?

### ISSUE THREAD

**Issue by jnbptstm** [patoarvizu/terraform-kubernetes-notary#5]
Hello,

I'm having this error while pushing and signing docker image :

```
> docker --debug push <my_registry>/<my_image>:<my_tag> 
The push refers to repository [<my_registry>/<my_image>]
7ad00cd55506: Layer already exists 
martin: digest: sha256:dcdf379c574e1773d703f0c0d56d67594e7a91d6b84d11ff46799f60fb081c52 size: 527
Signing and pushing trust metadata
DEBU[0000] reading certificate directory: /home/martin/.docker/tls/<my_notary_url> 
DEBU[0001] No yubikey found, using alternative key storage: no library found 
DEBU[0001] Making dir path: /home/martin/.docker/trust/tuf/<my_registry>/<my_image>/changelist 
DEBU[0001] received HTTP status 404 when requesting root. 
DEBU[0001] No yubikey found, using alternative key storage: no library found 
DEBU[0001] No yubikey found, using alternative key storage: no library found 
DEBU[0001] generated ECDSA key with keyID: <key_id> 
DEBU[0001] generated new ecdsa key for role: targets and keyID: <target_key_id> 
Error: trust data missing for remote repository <my_registry>/<my_image> or remote repository not found: timestamp key trust data unavailable.  Has a notary repository been initialized?

```

- Notary module version : 0.0.2
- Terraform version : 1.2.4
- Docker version : 18.06.1-ce 
- AWS EKS version : 1.19

### Terraform configuration

```
resource "aws_kms_key" "key" {}

resource "aws_kms_alias" "alias" {
  name = "alias/notary"
  target_key_id = aws_kms_key.key.id
}

resource "kubernetes_storage_class" "notary-storage-class" {
  metadata {
    name = "notary-storage-class"
  }
  storage_provisioner = "kubernetes.io/aws-ebs"
  reclaim_policy      = "Retain"
}

module "notary_generated-certificates" {
  source  = "patoarvizu/notary/kubernetes//modules/generated-certificates"
  version = "0.0.2"
  kms_alias_name = "notary"
  ecdsa_curve = "P256"
}

module "encrypted_server_db_password" {
  source =  "patoarvizu/notary/kubernetes//modules/encrypt-string"
  version = "0.0.2"
  text_to_encrypt = "password"
  kms_alias_name = "notary"
}

module "encrypted_signer_db_password" {
  source =  "patoarvizu/notary/kubernetes//modules/encrypt-string"
  version = "0.0.2"
  text_to_encrypt = "password"
  kms_alias_name = "notary"
}

module "encrypted_signer_alias_passphrase" {
  source =  "patoarvizu/notary/kubernetes//modules/encrypt-string"
  version = "0.0.2"
  text_to_encrypt = "password"
  kms_alias_name = "notary"
}

module "notary" {
  source  = "patoarvizu/notary/kubernetes"
  version = "0.0.2"
  namespace = "notary"

  ca_cert_pem = base64decode(module.notary_generated-certificates.ca_cert_pem)
  encrypted_server_cert_key = module.notary_generated-certificates.encrypted_server_cert_key
  encrypted_server_db_password = module.encrypted_server_db_password.encrypted_text
  encrypted_signer_alias_passphrase = module.encrypted_signer_alias_passphrase.encrypted_text
  encrypted_signer_cert_key = module.notary_generated-certificates.encrypted_signer_cert_key
  encrypted_signer_db_password = module.encrypted_signer_db_password.encrypted_text
  server_cert_pem = base64decode(module.notary_generated-certificates.server_cert_pem)
  signer_cert_pem = base64decode(module.notary_generated-certificates.signer_cert_pem)
  storage_class_name = kubernetes_storage_class.notary-storage-class.metadata[0].name

  server_replicas = 1
  signer_replicas = 1
  logging_level = "DEBUG"
  gun_prefixes = ["<my_registry>/"]
  storage_size = "5Gi"
  ingress_annotations = {
    "kubernetes.io/ingress.class" = "alb"
    "alb.ingress.kubernetes.io/scheme" = "internal"
    "alb.ingress.kubernetes.io/certificate-arn" = "<certificate_arn>"
    "alb.ingress.kubernetes.io/listen-ports" = "[{\"HTTP\":80}, {\"HTTPS\":443}]"
    "alb.ingress.kubernetes.io/actions.ssl-redirect" = "{'Type': 'redirect', 'RedirectConfig': { 'Protocol': 'HTTPS', 'Port': '443', 'StatusCode': 'HTTP_301'}}"
    "alb.ingress.kubernetes.io/group.name" = "ingress-notary"
    "alb.ingress.kubernetes.io/target-type" = "ip"
    "alb.ingress.kubernetes.io/healthcheck-port" = "65534"
    "alb.ingress.kubernetes.io/healthcheck-path" = "/actuator/health"
    "alb.ingress.kubernetes.io/backend-protocol" = "HTTPS"
  }
  ingress_path = "/*"
  ingress_hosts = ["<my_notary_url>"]
}
```

Kubernetes objects are well deployed.

### Local configuration

DOCKER_CONTENT_TRUST=1
DOCKER_CONTENT_TRUST_SERVER=https://<my_notary_url>

Docker trust directory (`~/.docker/trust/`) is empty.

### Logs

**notary-server** :

> {"level":"info","msg":"Version: 0.6.1, Git commit: d6e1431","time":"2022-07-12T16:17:20Z"}
> {"level":"info","msg":"Using remote signing service","time":"2022-07-12T16:17:20Z"}
> {"level":"info","msg":"Using mysql backend","time":"2022-07-12T16:17:20Z"}
> {"level":"info","msg":"Starting Server","time":"2022-07-12T16:17:20Z"}
> {"level":"info","msg":"Enabling TLS","time":"2022-07-12T16:17:20Z"}
> {"level":"info","msg":"Starting on :4443","time":"2022-07-12T16:17:20Z"}
> {"go.version":"go1.15.15","http.request.host":"<my_notary_url>","http.request.id":"54e083f6-842c-43c1-b84b-1dd80ae6bd4c","http.request.method":"GET","http.request.remoteaddr":"10.7.0.74","http.request.uri":"/v2/","http.request.useragent":"Docker-Client/18.06.1-ce (linux)","http.response.duration":"30.499µs","http.response.status":200,"http.response.written":2,"level":"info","msg":"response completed","time":"2022-07-12T16:43:24Z"}

**notary-signer** :

> {"level":"info","msg":"Version: 0.6.1, Git commit: d6e1431","time":"2022-07-12T16:17:17Z"}
> {"level":"debug","msg":"Default Alias: alias","time":"2022-07-12T16:17:17Z"}

Ingress redirects well to `notary-server` Service in HTTPS (last log line of `notary-server` is the response of `docker push`)
There seems to be a problem between `notary-server` and `notary-signer` because `notary-signer` does not log anything after the `docker push` (but maybe the logs are not really verbose).

I must be missing a configuration. Not sure if linked to Terraform module or Notary (opened issue : https://github.com/notaryproject/notary/issues/1646)

Does anyone have an idea ?
Thanks

**Comment by jnbptstm** [comment]
I think I found the issue.
In module's configuration, I have : `gun_prefixes = ["<my_registry_1>/", "<my_registry_2>/"]`
But this array is converted in ConfigMap  like `gun_prefixes = ["<my_registry_1>/, <my_registry_2>/"]` (one single string).
I've created a PR : #6

**Comment by patoarvizu** [comment]
Done in #6 and released in [v0.0.3](https://github.com/patoarvizu/terraform-kubernetes-notary/releases/tag/v0.0.3)!
